tests/api/cluster/test_app.py

```diff
@@ -206,6 +208,33 @@ def test_get_github_key_display_name(key, expected):
     assert cluster.App(None).get_github_key_display_name(key) == expected
 
 
+@patch("controlpanel.api.cluster.App.remove_auth_settings")
+@patch("controlpanel.api.cluster.App.get_deployment_envs")
+def test_delete_http_error(get_deployment_envs_mock, remove_auth_settings_mock, app):
+    app_cluster = cluster.App(app, github_api_token="testing")
+    get_deployment_envs_mock.side_effect = requests.exceptions.HTTPError()
+
+    app_cluster.delete()
+    assert remove_auth_settings_mock.call_count == 2
```

Comment:
I'd prob be more specific here
```suggestion
    remove_auth_settings_mock.assert_has_calls([
        call("dev"),
        call("prod"),
    ])
```